.gitlab/ci/docker/ubuntu2204_kokkos_cuda.dockerfile

```diff
@@ -53,14 +53,16 @@ RUN mkdir -p /opt/kokkos/build && \
     mkdir bld && cd bld && \
     CXX=/opt/kokkos/build/kokkos-$KOKKOS_VERSION/bin/nvcc_wrapper \
     cmake -B . -S ../kokkos-$KOKKOS_VERSION \
+          -DBUILD_SHARED_LIBS=ON \
```

Comment:
Kokkos sets BUILD_SHARED_LIBS=OFF by default (as in cmake's default behavior)

Reply:
It's weird that Kokkos compiles OK but this breaks our code.

Do you think the `-DCMAKE_CXX_FLAGS=-fPIC` is actually the problematic one?